components/server/src/analytics-controller.ts

```diff
@@ -177,4 +179,16 @@ export class AnalyticsController {
             throw e;
         }
     }
+
+    private setHashedUserIdCookie(userId: string, res: express.Response): void {
+        const hashedUserId = crypto.createHash("md5").update(userId).digest("hex");
+        const oneYearInSeconds = 365 * 24 * 60 * 60;
+        res.cookie("gitpod_hashed_user_id", hashedUserId, {
+            domain: ".gitpod.io",
```

Comment:
This should probably be the domain from the server config instead of hard-coded.

Reply:
> This should probably be the domain from the server config instead of hard-coded.

Why? we just wanted to compare across `*.gitpod.io` domains?

Reply:
It looks like we'll get errors about not being able to set the cookie because it's for a different domain than the one we're on.

Reply:
Raised https://github.com/gitpod-io/gitpod/pull/20229